### ISSUE TITLE
kurento telephone-event(2833) media transport

### ISSUE THREAD

**Issue by logcy** [Kurento/bugtracker#585]
## Prerequisites

These are MANDATORY, otherwise the issue will be automatically closed.

* [] I agree to fill this issue template.
* [] I have read the [Troubleshooting Guide] and [Support Instructions].

[Troubleshooting Guide]: https://doc-kurento.readthedocs.io/en/latest/user/troubleshooting.html
[Support Instructions]: https://github.com/Kurento/.github/blob/master/SUPPORT.md

## Issue description

Kurento cannot transmit two audio streams at the same time

## Context

When I try to use kurento to negotiate media with the other party, the other party will send two media types: pcma (pt 8) and telephone-event (2833) (pt 100) to negotiate with me. I configure telephone-event/ in Kurento. After the media type of 8000, the two media types of pcma and telephone/8000 can be negotiated with the other party, and the other party will transmit two m media streams, one is the media stream of Pcma, the other is the media stream of telephone-event, pcma The Pt used by the media stream is 8, and the pt used by the telephone-event is 100. At this time, kurento will report an error (Unknown PT: 100, expected: 8). How can I solve this problem?

### About Kurento Media Server

* Kurento version:  
* Server OS:          
* Installation method:
    
  - [] [apt-get]
  - [] [Docker]
  - [] [AWS CloudFormation]
  - [] [Built from sources]

[apt-get]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-local
[Docker]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-docker
[AWS CloudFormation]: https://doc-kurento.readthedocs.io/en/latest/user/installation.html#installation-aws
[Built from sources]: https://doc-kurento.readthedocs.io/en/latest/dev/dev_guide.html#dev-sources

### About your Application Server

* Programming Language:  
* Kurento Client version:     

### About end-user clients

* Device(s):    
* OS(es):        
* Browser(s):  

### Run these commands

```
cat /etc/lsb-release
```

```
kurento-media-server --version
```

```
dpkg -l | grep -Pi 'kurento|kms-|gst.*1.5|nice'
```

**Comment by j1elo** [comment]
please copy an example of SDP Offer that is sent to the Kurento endpoint

**Comment by logcy** [comment]
1、I config kurento with telephone-event/8000:
{
    "numAudioMedias": 1,
    "numVideoMedias": 1,
    "audioCodecs": [
        { "name": "PCMA/8000" },
        { "name": "telephone-event/8000" }
    ],
    "videoCodecs": [
        { "name": "H264/90000" }
    ]
}
in SdpEndpoint.conf.json
2、The Upstream server send the offer :
![image](https://user-images.githubusercontent.com/3400895/112566546-2a9a9600-8e1a-11eb-8a87-fa989b0a0cda.png)

3、then I reply the answer:
![image](https://user-images.githubusercontent.com/3400895/112566649-5c136180-8e1a-11eb-975c-eb44b10b8d5b.png)
4、When the call is connected，I will receive two streams in the audio transmission channel, one is the voice stream of pcma(with pt 8) and the other is the button stream of telephone-event(with pt 101)，then the kurento will give a error about "Unknown PT: 101, expected: 8"

**Comment by j1elo** [comment]
I don't think that will work because Kurento does not support more than 1 video and 1 audio negotiated on each Endpoint. You can always use an Endpoint for each audio, though.

On the other hand, you said:

> 2、The Upstream server send the offer
> 3、then I reply the answer

but who is the "I" in "*I reply the answer*"? Are you building an SDP Answer on your own, in your application, ignoring what Kurento says? Well if you ignore the reply from Kurento, it's not strange that things don't work.

You have to pass the SDP Offer to Kurento, and then use the Answer that you get back:

```
String sdpOffer = signaling.receiveOffer();
String sdpAnswer = kurentoEndpoint.processOffer(sdpOffer);
signaling.sendAnswer(sdpAnswer);
```

where "signaling" is your way of sending and receiving SDP messages between clients and the Kurento server.

**Comment by j1elo** [comment]
Now that I look at it a second time, I might understand now what you mean. There is in fact one audio media, but the payload in the RTP packets of this media might change dynamically, some times they could be 8, other times they could be 101.

This hasn't been tested before, because the main use case for Kurento is WebRTC comms, where the payload type doesn't change over time. So maybe the problem is that GStreamer (the library inside Kurento) gets the first PT, and configures itself for it, but Kurento won't accept more PT afterwards.

**Comment by logcy** [comment]
In the call center application, there is an IVR scenario, that is, the user navigates according to the phone buttons, and the application recognizes the user’s buttons through DTFM. There are three methods for DTMF transmission (In-band, RFC2833 (TeleponeEvent), SIPINFO). If you use RFC2833 (TeleponeEvent) For transmission, two pt-type streams pcma and TeleponeEvent will be transmitted on the same audio port. I have this problem now. Will you support this method of RTP streaming in the future? Is there any quick way I want to solve this problem now?

**Comment by kfbixc** [comment]
any progress？
